src/nimdowpkg/statusbar.nim

```diff
@@ -418,104 +407,110 @@ proc renderStringRightAligned(
         parsingSgr = true
         reset sgr
         continue
-    if not parsingCsi and font != nil:
-      let centerY = font.ascent + (this.area.height.int - font.height) div 2
-      XftDrawRect(this.draw, bgColor.addr, xLoc, 0, glyph.xOff.int, this.area.height.int)
-      XftDrawStringUtf8(
-        this.draw,
-        color.addr,
-        font,
-        xLoc,
-        centerY,
-        runeAddr,
-        rune.size
-      )
-      xLoc += glyph.xOff
-
+    let display = this.display
+    proc tryFont(font: PXftFont): bool =
+      if (not parsingCsi) and XftCharExists(display, font, rune.FcChar32) == 1:
+        var glyph: XGlyphInfo
+        XftTextExtentsUtf8(display, font, runeAddr, rune.size, glyph.addr)
+        runeInfo.add((rune, runeAddr, font, color, bgColor, glyph))
+        stringWidth += glyph.xOff
+        return true
+      if (not parsingCSI) and rune.int32 == 31:
+        runeInfo.add((rune, runeAddr, font, color, bgColor, default(XGlyphInfo))) # This is never rendered, but must be passed to parse clickables
+        return true
+    if selectedFont == -1:
+      for font in this.fonts:
+        if tryFont(font): break
+    else:
+      if selectedFont > this.fonts.high:
+        log "Unable to select font " & $selectedFont & ", too few colors defined in config", lvlError
```

Comment:
This will spam the log, need to only show this message once per bar render

Reply:
Users shouldn't really ever see this error unless they ask for a font they haven't set up, so I don't think it's going to be a huge issue for a little log spam in this case. But I added a flag to stop it from repeating too often, I guess we could also log the position in the string where the error occurred to be a little bit more helpful.